streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -407,7 +413,7 @@ public static StreamThread create(final InternalTopologyBuilder builder,
             referenceContainer.nextScheduledRebalanceMs,
             shutdownErrorHook,
             streamsUncaughtExceptionHandler,
-            cacheSize -> cache.resize(cacheSize)
```

Comment:
My ide tried to optimize this as well. At the time not passing in cacheSize caused some expections. I would be careful about making this change without need

Reply:
Interesting. It *should* be exactly the same, but of course who knows with Java. Did it cause a test to fail or was it something more subtle?

Reply:
I think it caused a test to fail but not everytime. It also could have been fixed since then as changes have been made. If all the tests pass it's probably fine